src/common/lru.zig

```diff
@@ -14,7 +14,6 @@ pub fn LruCache(comptime K: type, comptime V: type) type {
         dbl_link_list: TailQueue(LruEntry),
         max_items: usize,
         len: usize = 0,
-        mux: std.Thread.Mutex,
```

Comment:
change comment of 'thread-safe' LRU

Reply:
Hi @0xNineteen. I just stumbled upon this project few days back. It's an interesting project and Zig is a greattt language!

Just wondering...  why don't you use policy-based design to implement `LruCache` both as thread-safe and thread-unsafe? It's zero-cost to have this flexibility.

I mean, in this way:

```zig
const NullMutex = struct {
    const Self = @This();
    pub fn lock(_: *Self) void {}
    pub fn unlock(_: *Self) void {}
};

/// A thread-unsafe LRU Cache
pub fn LruCache(comptime K: type, comptime V: type) type {
    return LruCacheImpl(NullMutex, K, V);
}

/// A thread-safe LRU Cache
pub fn SafeLruCache(comptime K: type, comptime V: type) type {
    return LruCacheImpl(std.Thread.Mutex, K, V);
}

// private impl... 
fn LruCacheImpl(comptime Mutex: type, comptime K: type, comptime V: type) type {
      // the actual impl that uses Mutex as policy.
}
```

For the unsafe LruCache, the `NullMutex` lock/unlock calls would be optimized away (and removed) by the compiler.

Reply:
hey @snawaz - thanks for the comment! - rn we only use the LRU in the ping-cache which is already wrapped in a Mux so each LRU in the cache is already thread-safe using this approach

Reply:
Yes, you have mentioned that in the description itself, which I understood. I was just saying you get both thread-safe and thread-unsafe versions without paying any extra cost. It _could_ be useful in other scenarios (in the future) where you need thread-safety. Anyway, it's up to you. The code is otherwise great and I loved the project.

Reply:
yup agreed its another good soln which could be useful in future scenarios - thanks for the complements and the comment! :)